### DISCUSSION TITLE
Decomposing parts of source name

### DISCUSSION THREAD

**Discussion by johnkwaters** [dotnet/templating#6704] [Template authoring]
In our framework template we want to be able to use a syntax like Customer.Project as part of namespaces, filenames and directories, fpr instance creating a project file Customer.Project.Api with a namespace Customer.Project.Api etc. When using the template we would specify values for Customer.Project - currently done by using -n "SomeCustomer.SomeProject", and replacing the string Customer.Project. However, we also need the parts - customer and project - separately, for instance we want to create terraform templates that create azure resources like blobs and keyvaults, based on these two name parts, which need to be combined with different characters depending on the naming rules for each azure resource. For instance, a storage account might become customerprojectdevblob and a key vault customer-project-dev-kv.
What would be a good approach - and I don't think there is any specific terraform processing help, so would have to work with general text files?

**Comment by YuliiaKovalova** [comment]
Hi @johnkwaters,

We can recommend you create 2 separate symbols (parameters) and concat them together later by calling [join](https://github.com/dotnet/templating/wiki/Available-Symbols-Generators#join) macro.

**Reply by johnkwaters** [reply]
Thanks - so I can see how that will be helpful - but for instance, for filenames there is no way to do that - when instantiating the template, if we have a symbol mapping say -name SomeClient.SomeProject -> and all filenames that look like ClientName.ProjectName.XXX get renamed -> that doesn't work if we have two symbols, right? So we would need both the concatenated version in -name and the two parts? Because I can't set -name to be a concatenation, right?

**Reply by YuliiaKovalova** [reply]
Hi @johnkwaters,

You can use the concatenated _separate_ symbol and replace "ClientName.ProjectName.XXX" with. Also, probably regexes can help you with detecting a string that requires renaming.
Unfortunately `-name `param doesn't provide the required by your use case flexibility.
